test/test-helper.ts

```diff
@@ -211,3 +211,32 @@ export class TestDriver {
 
 export const onWindows = process.platform === "win32" ? it : it.skip;
 export const onPosix = process.platform !== "win32" ? it : it.skip;
+
+class Query {
+  private readonly stack: Record<string, any>;
+  constructor(stackInput: string) {
+    this.stack = JSON.parse(stackInput);
+  }
+
+  public byId(id: string): Record<string, any> {
+    const constructs = (
+      [
+        ...Object.values(this.stack.resource || {}),
+        ...Object.values(this.stack.data || {}),
+      ] as Record<string, any>[]
+    ).reduce(
+      (carry, item) => ({ ...carry, ...item }),
+      {} as Record<string, any>
+    );
```

Comment:
Maybe add a comment that this "jumps over" the resource type and combines all resources by id, ignoring their type? I had a hard time figuring out how this worked using `Object.values()` 😅 An alternative could be adding stronger types, which I'd probably go for.
I'm thinking about something like (with better names 😄):
```ts
type RenderedTerraformResource = Record<string, any>;
type TerraformProvider = string;
type Id = string;
type RenderedStack = {
  resource: Record<TerraformProvider, Record<Id, RenderedTerraformResource>>,
  data: Record<TerraformProvider, Record<Id, RenderedTerraformResource>>,
  output: Record<Id, {value: any}>
};
const constructs: RenderedTerraformResource[];
```
(This is probably too verbose 😂 But I hope I could convey what I meant)